subprojects/docs/src/samples/maven-publish/conditional-publishing/build.gradle

```diff
@@ -0,0 +1,70 @@
+apply plugin: 'java'
+apply plugin: 'maven-publish'
+
+group = 'org.gradle.sample'
+version = '1.0'
+
+task sourcesJar(type: Jar) {
+    classifier = 'sources'
+    from sourceSets.main.allJava
+}
+
+// START SNIPPET publishing
+publishing {
+    publications {
+        binary(MavenPublication) {
+            from components.java
+        }
+        binaryAndSources(MavenPublication) {
+            from components.java
+            artifact sourcesJar
+        }
+    }
+    repositories {
+        // change URLs to point to your repos, e.g. http://my.org/repo
+        maven {
+            name "external"
+            url "$buildDir/repos/external"
+        }
+        maven {
+            name "internal"
+            url "$buildDir/repos/internal"
+        }
+    }
+}
+// END SNIPPET publishing
+
+// START SNIPPET task-config
+tasks.withType(PublishToMavenRepository) {
+    onlyIf {
+        if (repository == publishing.repositories.external)
+            return publication == publishing.publications.binary
+        if (repository == publishing.repositories.internal)
+            return publication == publishing.publications.binaryAndSources
+        return false
```

Comment:
Would a `switch` be more appropriate here?

Reply:
I don't think it would add clarity but would mean more lines, so I think keeping the `if` should be okay for now.